inspire_utils/name.py

```diff
@@ -0,0 +1,234 @@
+
+    def __str__(self):
+        return str(self.parsed_name)
+
+    @classmethod
+    def loads(cls, name):
+        """Load a parsed name from a string.
+
+        Raises:
+            TypeError: when name isn't a type of `six.string_types`.
+            ValueError: when name is empty or None.
+        """
+        if not isinstance(name, six.string_types):
+            raise TypeError('arguments to {classname} must be of type {string_types}'.format(
+                classname=cls.__name__, string_types=repr(six.string_types)
+            ))
+        if not name or name.isspace():
+            raise ValueError('name must not be empty')
+
+        return cls(name)
+
+    def dumps(self):
+        """Dump the name to string, after normalizing it."""
+        def _is_initial(author_name):
+            return len(author_name) == 1 or u'.' in author_name
+
+        def _ensure_dotted_initials(author_name):
+            if _is_initial(author_name) \
+                    and u'.' not in author_name:
+                seq = (author_name, u'.')
+                author_name = u''.join(seq)
+            return author_name
+
+        def _ensure_dotted_suffixes(author_suffix):
+            if u'.' not in author_suffix:
+                seq = (author_suffix, u'.')
+                author_suffix = u''.join(seq)
+            return author_suffix
+
+        def _is_roman_numeral(suffix):
+            """Controls that the user's input only contains valid roman numerals"""
+            valid_roman_numerals = [u'M', u'D', u'C', u'L', u'X',
+                                    u'V', u'I', u'(', u')']
+            return all(letters in valid_roman_numerals
+                       for letters in suffix.upper())
+
+        # Create first and middle
+        first_name = _ensure_dotted_initials(self.parsed_name.first)
+        middle_name = _ensure_dotted_initials(self.parsed_name.middle)
+
+        if _is_initial(first_name) and _is_initial(middle_name):
+            normalized_names = u'{first_name}{middle_name}'
+        else:
+            normalized_names = u'{first_name} {middle_name}'
+
+        normalized_names = normalized_names.format(
+            first_name=first_name,
+            middle_name=middle_name,
+        )
+
+        if _is_roman_numeral(self.parsed_name.suffix):
+            suffix = self.parsed_name.suffix.upper()
+        else:
+            suffix = _ensure_dotted_suffixes(self.parsed_name.suffix)
+
+        final_name = u', '.join(
+            part for part in (self.parsed_name.last, normalized_names.strip(), suffix)
+            if part)
+
+        # Replace unicode curly apostrophe to normal apostrophe.
+        final_name = final_name.replace(u'’', '\'')
+
+        return final_name
+
+
+def normalize_name(name):
+    """Normalize name.
+
+    Args:
+        name (six.text_type): The name to be normalized.
+
+    Returns:
+        str: The normalized name.
+    """
+    return ParsedName.loads(name).dumps()
```

Comment:
here it would be good to return `None` when `name is None`, as for `normalize_date`: https://github.com/inspirehep/inspire-utils/blob/35181393b34d098af044a21a1b4041803f210fed/inspire_utils/date.py#L227-L228 (the reason is that it can then be used directly in the builders whithout having to check for `None` there.

Reply:
Good catch! 
I've added [this check](https://github.com/inspirehep/inspire-utils/pull/18/files#diff-f116d82c04785f5d37d57c86cae932cdR158), which ensures not getting an empty (whitespace only) name also.